Source/WebKit/Shared/Extensions/WebExtensionWindow.serialization.in

```diff
@@ -0,0 +1,59 @@
+# Copyright (C) 2023 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#if ENABLE(WK_WEB_EXTENSIONS)
+
+headers: "WebExtensionWindow.h"
+
+struct WebKit::WebExtensionWindowParameters {
+    std::optional<WebKit::WebExtensionWindowIdentifier> identifier;
+
+    std::optional<WebKit::WebExtensionWindow::State> state;
+    std::optional<WebKit::WebExtensionWindow::Type> type;
+
+    std::optional<Vector<WebKit::WebExtensionTabParameters>> tabs;
+
+    std::optional<CGRect> frame;
+
+    std::optional<bool> focused;
+    std::optional<bool> privateBrowsing;
+};
+
+[Nested] enum class WebKit::WebExtensionWindow::PopulateTabs : bool;
```

Comment:
Does ` : bool` mean we will get PopulateTabs::Yes and PopulateTabs::No?

Reply:
Yeah, the `: bool` is just the storage type.